web-ui/src/pages/FeedbackRequestPage.jsx

```diff
@@ -125,78 +161,98 @@ const FeedbackRequestPage = () => {
                   Back
                 </Button>
               </Link>
 
               <Link
-                className={`no-underline-link ${activeStep > getSteps().length ? 'disabled-link no-underline-link' : ''}`}
-                to={activeStep===3 ?`/feedback/request/confirmation` : getFeedbackArgs(activeStep + 1)}>
+                  className={`no-underline-link ${activeStep > getSteps().length ? 'disabled-link no-underline-link' : ''}`}
+                  to={activeStep === 3 ? `/feedback/request/confirmation` : `?step=${activeStep + 1}`}>
                 <Button
-                  disabled={activeStep > getSteps().length}
-                  variant="contained"
-                  color="primary">
+                    disabled={activeStep > getSteps().length}
+                    variant="contained"
+                    color="primary">
                   {activeStep === steps.length ? "Submit" : "Next"}
                 </Button>
               </Link>
             </div>
+          </div>
         </div>
-      </div>
-      <Stepper activeStep={activeStep - 1} className={classes.root}>
-        {steps.map((label) => {
-          const stepProps = {};
-          const labelProps = {};
-          return (
-            <Step key={label} {...stepProps}>
-              <StepLabel {...labelProps} key={label}>{label}</StepLabel>
-            </Step>
-          );
-        })}
-      </Stepper>
-      <div className="current-step-content">
+        <Stepper activeStep={activeStep - 1} className={classes.root}>
+          {steps.map((label) => {
+            const stepProps = {};
+            const labelProps = {};
+            return (
+                <Step key={label} {...stepProps}>
+                  <StepLabel {...labelProps} key={label}>{label}</StepLabel>
+                </Step>
+            );
+          })}
+        </Stepper>
+
+        <div className="current-step-content">
           {activeStep === 1 && <TemplatePreviewModal/> &&
-          <div className="card-container">
-            {getTemplates().map((template) => (
-              <TemplateCard
-                title={template.title}
-                creator={template.creator}
-                description={template.description}
-                isAdHoc={template.isAdHoc}
-                questions={template.questions}
-                expanded={preview.open}
-                onClick={(e) => handlePreviewOpen(e, template)}
-                onCardClick={() => onCardClick(template)}/>
-            ))}
+          <div className="search-bar">
+            <TextField
+                label="Search Templates..."
+                placeholder="Template 1"
+                value={searchText}
+                onChange={(e) => {
+                  setSearchText(e.target.value);
+                }}
+            />
+            <div className="card-container">
+              {
+                (filteredTemplates.length === 0)
+                    ? <h2>No matching templates</h2>
```

Comment:
This renders even if there are no templates to display, regardless of whether the user has entered any search text. Maybe something like:
```
{searchText === "" && filteredTemplates.length === 0 &&
<h2>No templates found</h2>}
{searchText && searchText.length > 0 && filteredTemplates.length === 0 &&
<h2>No matching templates</h2>}